posts/telemetry-yasmin.adoc

```diff
@@ -0,0 +1,253 @@
+    <version>1.19.0</version>
+</dependency>
+<dependency>
+    <groupId>io.opentelemetry.instrumentation</groupId>
+    <artifactId>opentelemetry-instrumentation-annotions</artifactId>
+    <version>1.19.0-alpha</version>
+</dependency>
+----
+
+### Code Instrumentation
+
+Now you are all set to instrument your code to manually create traces. We start by link:https://github.com/yasmin-aumeeruddy/mpTelemetry-Demo/blob/main/inventory/src/main/java/io/openliberty/demo/inventory/InventoryResource.java#L38-L42[`injecting a Tracer and Span`] object: 
+
+[source, java]
+----
+@Inject
+Tracer tracer;
+
+@Inject
+Span getPropertiesSpan;
+----
+
+This can then be used to create spans. For example, a span called `GettingProperties` starts and an event is added before the `system` service is called: 
+
+[source,java]
+----
+getPropertiesSpan = tracer.spanBuilder("GettingProperties").startSpan();
+Properties props = manager.get(hostname);
+try(Scope scope = getPropertiesSpan.makeCurrent()){
+    ...
+    getPropertiesSpan.addEvent("Received properties");
+}
+finally{
+    getPropertiesSpan.end();
+}
+----
+
+You can also create new spans by annotating methods in any Jakarta CDI beans using link:https://opentelemetry.io/docs/instrumentation/java/automatic/annotations/[`@WithSpan`]. link:https://github.com/yasmin-aumeeruddy/mpTelemetry-Demo/blob/main/inventory/src/main/java/io/openliberty/demo/inventory/InventoryManager.java#L47-L48[Here] a span is created when a new system is added to the inventory. The `hostname` method parameter is annotated with the `@SpanAttribute` annotation to indicate that it is part of the trace: 
+
+[source, java]
+----
+@WithSpan
+public void add(@SpanAttribute(value = "hostname") String hostname, Properties systemProps) {
+    ...
+}
+----
+
+A span created using the `@WithSpan` annotation can be given a name. For example, the span created link: https://github.com/yasmin-aumeeruddy/mpTelemetry-Demo/blob/main/inventory/src/main/java/io/openliberty/demo/inventory/InventoryManager.java#L58[here] is given the name "list":
+
+[source,java]
+----
+@WithSpan("list")
+public InventoryList list() {
+    ...
+}
+----
+
+Now that spans have been created manually, we can build the inventory service and deploy it to Open Liberty:
+
+`cd inventory`
+
+`mvn liberty:run`
+
+Navigate to the inventory endpoint:
+
+`http://localhost:9081/inventory/systems/localhost`
+
+You should see five spans in the exporter's endpoint: four spans from inventory and one span from system. For example:
+
+image::img/blog/mptelemetry_inventory_manual_span.png[Inventory manual span]
+
+These spans are all part of one single trace which is emitted from the request to the endpoint above. It is therefore easier to identify the source of inaccuracy or latency in a single request by debugging spans individually.  
+
+See link:https://opentelemetry.io/docs/instrumentation/java/manual/[OpenTelemetry's manual instrumentation documentation].
+
+## Java Agent
+
```

Comment:
what are the advantages of using the agent? Why might someone want do it? The reason for manual instrumentation is clear, but not so much why you'd want to use the agent.

Reply:
I have updated the java agent introduction